### ISSUE TITLE
Exception on Group

### ISSUE THREAD

**Issue by ASudre** [jenkinsci/oic-auth-plugin#49]
Can't find a way to send groups :

- When using an stringified array, we get this error :
`java.lang.String cannot be cast to java.util.List`
On the line :
https://github.com/jenkinsci/oic-auth-plugin/blob/209cf6e212ca6d8e4dc14b8f4e5867203c55d34d/src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java#L473

- When using an array, we get this error :
```com.google.api.client.auth.oauth2.TokenResponseException: 500 Internal Server Error
{
  "error" : "Internal Error"
}
    at com.google.api.client.auth.oauth2.TokenResponseException.from(TokenResponseException.java:105)
    at com.google.api.client.auth.oauth2.TokenRequest.executeUnparsed(TokenRequest.java:287)
    at com.google.api.client.auth.openidconnect.IdTokenResponse.execute(IdTokenResponse.java:120)
    at org.jenkinsci.plugins.oic.OicSecurityRealm$3.onSuccess(OicSecurityRealm.java:328)...
```

Error with the JSON parser ? Or the cast on line 473 (v4) ? Or from our part ?

**Comment by mjmbischoff** [comment]
I'd like to make a test case for this to make sure I fix the issue and prevent regressions.

@ASudre Are you able you supply an example response?

**Comment by GregoireW** [comment]
I got the same error on some profile in my company.  I use a certified openid provider, and it has a bad way of managing some profile. 

- If the profile has multiple groups, no problem, the solution is working correctly  (  "groups": ["a", "b"] ) 
- If the profile has 1 group, instead of returning an array, it returns a single string ( "groups": "a" ) . I face the `java.lang.String cannot be cast to java.util.List` error.   I started to check the openid / jwt RFC to check if there is something about list of single value transformed to simple data, but I did not find anything about this. 

- We try to change the way our provider returned data and we even end up with something like:  "groups": "[a,b]"   which is worst. 
We end up putting a rule in the provider profile: If group is size 1 then add "" to group. 

@ASudre perhaps you can check if you have the same thing with profile with single group / multiple groups.

@mjmbischoff  I'm not sure if there is a lot of oidc providers with this 'feature', so i'm not sure if I shoud make a PR with a change   like  "if group is string then ... else if group is list then ...  else groups not found  "   What do you think ?

**Comment by ilkomiliev** [comment]
hi, we are using AWS Cognito as OIDC and custom attribute mappings to pass the roles. Unfortunately, Cognito supports only two types for the mapped attributes - string and number. So, the array of the roles is tostringed() in the process of this mapping, causing the ClassCastException :-( 

@ASudre: I'm getting this error in case where the OIDC provider already has returned internal error. In my case I've experimented altering the Cognito response by the provided triggers and as soon as I messed something in the triggered lambda I got this error message. Perhaps you can check, if this response is not a follow up of previous error on the OIDC side.

**Comment by lllmaa** [comment]
this might be related to the issue @GregoireW mentioned:
https://issues.jenkins-ci.org/browse/JENKINS-61609
PR for fix in:
https://github.com/jenkinsci/oic-auth-plugin/pull/90

**Comment by ArbazSheikh** [comment]
while working with WSO2 5.10 openid provider, even the ( "groups": ["a", "b"] ) format in the token is giving
`java.lang.ClassCastException: class java.lang.String cannot be cast to class java.util.List (java.lang.String and java.util.List are in module java.base of loader 'bootstrap')` error

**Comment by michael-doubez** [comment]
Closed by #122
